### ISSUE TITLE
Release drag when pointer get out of element

### ISSUE THREAD

**Issue by jeiea** [davidcetinkaya/embla-carousel#101]
# What I did
1. Open this [example](https://codesandbox.io/s/embla-carousel-default-vanilla-gqh0n).
2. Press mouse left button in embla.
3. Drag out of right pane (browser page)
4. Release the button.
5. Move the pointer to right pane.

# What I expected
Smooth scroll like [keen-slider](https://codesandbox.io/s/keen-slider-free-react-v520u?file=/src/App.js).

# What I got
Embla captures mouse in whole page, and don't reset press status when pointer get out of page.

**Comment by davidcetinkaya** [comment]
Hello @jeiea,
Thank you for the clear issue description, I appreciate that 👍.

I'm not sure there are any obvious benefits with mimicking **keen-slider** here. Here's why in my opinion:

- Not tracking the pointer when it has moved outside of the element bounds (assuming the pointer down event happened inside the element), we might miss important changes in how fast the user is dragging the carousel.
- Some use cases may be carousels that doesn't take up the full width/height of the window. In these cases only tracking the pointer inside the carousel element would make it hard to drag in a predictable way, if the slightest drag outside the element bounds were to stop tracking the pointer (because the carousel has a small area on the webpage, its draggable area would also be small).
- How do we know if dragging the pointer outside of the element was intentional? Please take a look at [this video](https://streamable.com/u9zna9) where I'm dragging the carousel with a bit of force, but I don't intend to actually move the cursor outside of the carousel element.

The example [CodeSandbox link](https://codesandbox.io/s/embla-carousel-default-vanilla-gqh0n) you mention is sandboxed in an **iframe**, so I understand that the pointer gets stuck when dragging outside the iframe bounds, but if you open up the same CodeSandbox in full view without the iframe [here](https://gqh0n.csb.app/), you'll see that leaving the browser window with the pointer is working as expected.

To sum up, I don't think this issue should be prioritized among other things that could be improved with Embla Carousel.

**Comment by jeiea** [comment]
Okay, that sounds reasonable. Carousel have the reason that tracking a pointer out of the element.

I still feel wierd that a carousel tracks a pointer with no pressed buttons but you informed to me that it is limited to iframe usage. I don't think this case is common so I agree that the priority is lower.

Feel free to close this issue.

**Comment by davidcetinkaya** [comment]
@jeiea I agree that it feels a bit weird. But at the time of writing I don't see an optimal solution for this. Flickity has been around for some time now and it works the same way as Embla when it comes to tracking the pointer outside of the carousel element (see [this CodeSandbox](https://codesandbox.io/s/react-flickity-w85iq) as a Flickity example).

I'm guessing a bit here but if the iframe behavior with the pointer would be considered a serious issue, Flickity would have payed attention to it by now, considering the fact that people actually pay for using it commercially.

Nonetheless, thank you for your suggestion.
